### PR TITLE
IDEX-3913: Implement stack api

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ok

**Comment by riuvshin** [comment]
Not OK, please read comment here https://github.com/codenvy/che/pull/216#issuecomment-177157583

**Comment by AndrienkoAleksandr** [comment]
eclipse/che#216 (comment) Fixed

File: onpremises-ide-packaging-tomcat-api/src/assembly/assembly.xml [hunk 1]
```diff
@@ -31,6 +31,19 @@
...
         <dependencySet>
             <useProjectArtifact>false</useProjectArtifact>
+            <unpack>true</unpack>
+            <outputDirectory>stacks</outputDirectory>
+            <includes>
+                <include>org.eclipse.che.core:che-core-ide-stacks</include>
+            </includes>
+            <unpackOptions>
+                <excludes>
+                    <exclude>META-INF/**</exclude>
```
Comment (skabashnyuk) [inline onpremises-ide-packaging-tomcat-api/src/assembly/assembly.xml]: why do you need that?

Comment (AndrienkoAleksandr) [inline onpremises-ide-packaging-tomcat-api/src/assembly/assembly.xml]: StackLoader loads predefined stacks from {catalina.base}/stacks/predefined-stacks.json

**Comment by riuvshin** [comment]
OK

**Pull request merged**